src/libraries/System.Private.CoreLib/src/System/UnreachableException.cs

```diff
@@ -0,0 +1,23 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace System
+{
+    public sealed class UnreachableException : SystemException
```

Comment:
@bartonjs, the notes are wishy-washy on the base type. Who's on the hook for deciding whether the exception will get "special treatment"?

Reply:
In https://github.com/dotnet/runtime/issues/35324#issuecomment-896330470 Jan pointed out that the existing special exceptions are differently special than we (I) remembered during the discussion.  He seems strongly against being special, so the base type should probably just be Exception.